nixos/roles/k3s/server.nix

```diff
@@ -0,0 +1,240 @@
+      ensureUsers = [ {
+        name = "root";
+        ensurePermissions = {
+          "DATABASE kubernetes" = "ALL PRIVILEGES";
+        };
+      } ];
+    };
+
+    flyingcircus.services.sensu-client.checks = let
+    in
+    {
+
+      cluster-dns = {
+        notification = "Cluster DNS (CoreDNS) is not healthy";
+        command = ''
+          ${pkgs.monitoring-plugins}/bin/check_http -j HEAD -H ${netCfg.clusterDns} -p 9153 -u /metrics
+        '';
+      };
+
+      kube-apiserver = {
+        notification = "Kubernetes API server is not working";
+        command = ''
+          ${pkgs.monitoring-plugins}/bin/check_http -j HEAD -H localhost -p 10251 -u /metrics
+        '';
+      };
+
+      kube-dashboard-metrics-scraper = {
+        notification = "Kubernetes dashboard metrics scraper sidecar is not working";
+        command = ''
+          ${pkgs.monitoring-plugins}/bin/check_http -H localhost -p 8000 -u /healthz
+        '';
+      };
+
+      kube-dashboard = {
+        notification = "Kubernetes dashboard backend is not working";
+        command = ''
+          ${pkgs.monitoring-plugins}/bin/check_http -H localhost -p 11000 -u /api/v1/namespace
+        '';
+      };
+
+    };
+
+    networking.nameservers = lib.mkOverride 90 (lib.take 3 ([netCfg.clusterDns] ++ fcNameservers));
+
+    services.k3s = let
+      k3sFlags = [
+        "--cluster-cidr=${netCfg.podCidr}"
+        "--service-cidr=${netCfg.serviceCidr}"
+        "--cluster-dns=${netCfg.clusterDns}"
+        "--node-ip=${nodeAddress}"
+        "--write-kubeconfig=${defaultKubeconfig}"
+        "--node-taint=node-role.kubernetes.io/server=true:NoSchedule"
+        "--flannel-backend=host-gw"
+        "--flannel-iface=ethsrv"
+        "--datastore-endpoint=postgres://@:5432/kubernetes?host=/run/postgresql"
+        "--token-file=/var/lib/k3s/secret_token"
+        "--data-dir=/var/lib/k3s"
+        "--kube-apiserver-arg enable-admission-plugins=PodNodeSelector"
+      ];
+    in {
+      enable = true;
+      role = "server";
+      inherit tokenFile;
+      extraFlags = lib.concatStringsSep " " k3sFlags;
+    };
+
+    systemd.services.fc-set-k3s-config-permissions = {
+      requires = [ "k3s.service" ];
+      partOf = [ "k3s.service" ];
+      wantedBy = [ "k3s.service" ];
+      after = [ "k3s.service" ];
+      path = [ pkgs.acl ];
+      script = ''
+        echo "Grant sudo-srv access to k3s config file..."
+        setfacl -m g:sudo-srv:r ${defaultKubeconfig}
+        echo "Grant kubernetes user access to k3s config file..."
+        setfacl -m u:kubernetes:r ${defaultKubeconfig}
+      '';
+      serviceConfig = {
+        RemainAfterExit = true;
+        Type = "oneshot";
+      };
+    };
+
+    ### Dashboard
+    flyingcircus.services.nginx.enable = true;
+
+    services.nginx.virtualHosts = {
+      "${head addresses}" = {
+        enableACME = true;
+        serverAliases = tail addresses;
+        extraConfig = ''
+          auth_basic "FCIO";
```

Comment:
vpn without basic auth + option flag to  completely prohibit access from outside

Reply:
and use the .sudo-srv file